internal/provider/adc/translator/apisixconsumer.go

```diff
@@ -0,0 +1,360 @@
+)
+
+var (
+	_errKeyNotFoundOrInvalid      = errors.New("key \"key\" not found or invalid in secret")
+	_errUsernameNotFoundOrInvalid = errors.New("key \"username\" not found or invalid in secret")
+	_errPasswordNotFoundOrInvalid = errors.New("key \"password\" not found or invalid in secret")
+
+	_jwtAuthExpDefaultValue = int64(868400)
+
+	_hmacAuthAlgorithmDefaultValue           = "hmac-sha256"
+	_hmacAuthClockSkewDefaultValue           = int64(0)
+	_hmacAuthKeepHeadersDefaultValue         = false
+	_hmacAuthEncodeURIParamsDefaultValue     = true
+	_hmacAuthValidateRequestBodyDefaultValue = false
+	_hmacAuthMaxReqBodyDefaultValue          = int64(524288)
+
+	_stringTrue = "true"
+)
+
+func (t *Translator) TranslateApisixConsumer(tctx *provider.TranslateContext, ac *v2.ApisixConsumer) (*TranslateResult, error) {
+	result := &TranslateResult{}
+	plugins := make(adctypes.Plugins)
+	if ac.Spec.AuthParameter.KeyAuth != nil {
+		cfg, err := t.translateConsumerKeyAuthPluginV2(tctx, ac.Namespace, ac.Spec.AuthParameter.KeyAuth)
+		if err != nil {
+			return nil, fmt.Errorf("invalid key auth config: %s", err)
+		}
+		plugins["key-auth"] = cfg
+	} else if ac.Spec.AuthParameter.BasicAuth != nil {
+		cfg, err := t.translateConsumerBasicAuthPluginV2(tctx, ac.Namespace, ac.Spec.AuthParameter.BasicAuth)
+		if err != nil {
+			return nil, fmt.Errorf("invalid basic auth config: %s", err)
+		}
+		plugins["basic-auth"] = cfg
+	} else if ac.Spec.AuthParameter.JwtAuth != nil {
+		cfg, err := t.translateConsumerJwtAuthPluginV2(tctx, ac.Namespace, ac.Spec.AuthParameter.JwtAuth)
+		if err != nil {
+			return nil, fmt.Errorf("invalid jwt auth config: %s", err)
+		}
+		plugins["jwt-auth"] = cfg
+	} else if ac.Spec.AuthParameter.WolfRBAC != nil {
+		cfg, err := t.translateConsumerWolfRBACPluginV2(tctx, ac.Namespace, ac.Spec.AuthParameter.WolfRBAC)
+		if err != nil {
+			return nil, fmt.Errorf("invalid wolf rbac config: %s", err)
+		}
+		plugins["wolf-rbac"] = cfg
+	} else if ac.Spec.AuthParameter.HMACAuth != nil {
+		cfg, err := t.translateConsumerHMACAuthPluginV2(tctx, ac.Namespace, ac.Spec.AuthParameter.HMACAuth)
+		if err != nil {
+			return nil, fmt.Errorf("invalid hmac auth config: %s", err)
+		}
+		plugins["hmac-auth"] = cfg
+	} else if ac.Spec.AuthParameter.LDAPAuth != nil {
+		cfg, err := t.translateConsumerLDAPAuthPluginV2(tctx, ac.Namespace, ac.Spec.AuthParameter.LDAPAuth)
+		if err != nil {
+			return nil, fmt.Errorf("invalid ldap auth config: %s", err)
+		}
+		plugins["ldap-auth"] = cfg
+	}
+
+	username := adctypes.ComposeConsumerName(ac.Namespace, ac.Name)
+	consumer := &adctypes.Consumer{
+		Username: username,
+	}
+	consumer.Plugins = plugins
+	consumer.Labels = label.GenLabel(ac)
+	result.Consumers = append(result.Consumers, consumer)
+	return result, nil
+}
+
+func (t *Translator) translateConsumerKeyAuthPluginV2(tctx *provider.TranslateContext, consumerNamespace string, cfg *v2.ApisixConsumerKeyAuth) (*types.KeyAuthConsumerConfig, error) {
```

Comment:
我想这些函数名可以不加 ”V2“ 了，用 ”ApisixConsumer“ 标识就可以了
```suggestion
func (t *Translator) translateApisixConsumerKeyAuth(tctx *provider.TranslateContext, consumerNamespace string, cfg *v2.ApisixConsumerKeyAuth) (*types.KeyAuthConsumerConfig, error) {
```

Reply:
fixed